xml/System.Text/Rune.xml

```diff
@@ -33,7 +33,196 @@
+
+The following sections explain:
+
+* [When to use the Rune type](#when-to-use-the-rune-type)
+* [When not to use the Rune type](#when-not-to-use-rune)
+* [How to instantiate a Rune](#how-to-instantiate-a-rune)
+* [How to query properties of a Rune instance](#query-properties-of-a-rune)
+* [Convert a `Rune` to UTF-8 or UTF-16](#convert-a-rune-to-utf-8-or-utf-16)
+* [Rune in .NET vs. other languages](#rune-in-net-vs-other-languages)
+
+### When to use the Rune type
+
+Consider using the `Rune` type if your code:
+
+* Calls APIs that require Unicode scalar values
+* Explicitly handles surrogate pairs
+
+#### APIs that require Unicode scalar values
+
+If your code iterates through the `char` instances in a `string` or a `ReadOnlySpan<char>`, some of the `char` methods won't work correctly on `char` instances that are in the surrogate range. For example, the following APIs require a scalar value `char` to work correctly:
+
+* <xref:System.Char.GetNumericValue%2A?displayProperty=nameWithType>
+* <xref:System.Char.GetUnicodeCategory%2A?displayProperty=nameWithType>
+* <xref:System.Char.IsDigit%2A?displayProperty=nameWithType>
+* <xref:System.Char.IsLetter%2A?displayProperty=nameWithType>
+* <xref:System.Char.IsLetterOrDigit%2A?displayProperty=nameWithType>
+* <xref:System.Char.IsLower%2A?displayProperty=nameWithType>
+* <xref:System.Char.IsNumber%2A?displayProperty=nameWithType>
+* <xref:System.Char.IsPunctuation%2A?displayProperty=nameWithType>
+* <xref:System.Char.IsSymbol%2A?displayProperty=nameWithType>
+* <xref:System.Char.IsUpper%2A?displayProperty=nameWithType>
+
+The following examples show code that won't work correctly if any of the `char` instances are surrogate code points:
+
+:::code language="csharp" source="~/snippets/System.Text/Rune/csharp/CountLettersInString.cs" id="SnippetBadExample":::
+
+:::code language="csharp" source="~/snippets/System.Text/Rune/csharp/CountLettersInSpan.cs" id="SnippetBadExample":::
+
+The preceding code works correctly with some languages such as English:
+
+```csharp
+CountLettersInString("Hello")
+// Returns 5
+```
+
+But it won't work correctly for languages outside the Basic Multilingual Plane, such as Osage:
+
+```csharp
+CountLettersInString("𐓏𐓘𐓻𐓘𐓻𐓟 𐒻𐓟")
+// Returns 0
+```
+
+The reason this method returns incorrect results for Osage text is that the `char` instances for Osage letters are surrogate code points. No single surrogate code point has enough information to determine if it's a letter.
+
+If you change this code to use `Rune` instead of `char`, the method works correctly with code points outside the Basic Multilingual Plane:
+
+:::code language="csharp" source="~/snippets/System.Text/Rune/csharp/CountLettersInString.cs" id="SnippetGoodExample":::
+
+:::code language="csharp" source="~/snippets/System.Text/Rune/csharp/CountLettersInSpan.cs" id="SnippetGoodExample":::
```

Comment:
I think it's good practice that when you have two snippets, one after another, they should have an introductory sentence between them for better separation. Otherwise it's pretty ugly to have two giant boxes drawn next to each other.

There are some other places that are squished together like this one.